src/gui/accountstate.cpp

```diff
@@ -113,42 +113,45 @@ AccountState::AccountState(AccountPtr account)
-                    setState(State::Disconnected);
-                }
-                [[fallthrough]];
-            case QNetworkInformation::Reachability::Local:
-                break;
+    connect(NetworkInformation::instance(), &NetworkInformation::reachabilityChanged, this, [this](NetworkInformation::Reachability reachability) {
+        switch (reachability) {
+        case NetworkInformation::Reachability::Online:
+            [[fallthrough]];
+        case NetworkInformation::Reachability::Site:
+            [[fallthrough]];
+        case NetworkInformation::Reachability::Unknown:
+            // the connection might not yet be established
+            QTimer::singleShot(0, this, [this] { checkConnectivity(false); });
+            break;
+        case NetworkInformation::Reachability::Disconnected:
+            // explicitly set disconnected, this way a successful checkConnectivity call above will trigger a local discover
+            if (state() != State::SignedOut) {
+                setState(State::Disconnected);
             }
-        });
+            [[fallthrough]];
+        case NetworkInformation::Reachability::Local:
+            break;
+        }
+    });
 
-        connect(qNetInfo, &QNetworkInformation::isMeteredChanged, this, [this](bool isMetered) {
-            if (ConfigFile().pauseSyncWhenMetered()) {
-                if (state() == State::Connected && isMetered) {
-                    qCInfo(lcAccountState) << "Network switched to a metered connection, setting account state to PausedDueToMetered";
-                    setState(State::PausedDueToMetered);
-                } else if (state() == State::PausedDueToMetered && !isMetered) {
-                    qCInfo(lcAccountState) << "Network switched to a NON-metered connection, setting account state to Connected";
-                    setState(State::Connected);
-                }
+    connect(NetworkInformation::instance(), &NetworkInformation::isMeteredChanged, this, [this](bool isMetered) {
+        if (ConfigFile().pauseSyncWhenMetered()) {
+            if (state() == State::Connected && isMetered) {
+                qCInfo(lcAccountState) << "Network switched to a metered connection, setting account state to PausedDueToMetered";
+                setState(State::Connecting);
+            } else if (state() == State::Connecting && !isMetered) {
+                qCInfo(lcAccountState) << "Network switched to a NON-metered connection, setting account state to Connected";
+                setState(State::Connected);
             }
-        });
-    }
-#endif
+        }
+    });
+
+    connect(NetworkInformation::instance(), &NetworkInformation::isBehindCaptivePortalChanged, this, [this](bool) {
+        // A direct connect is not possible, because then the state parameter of `isBehindCaptivePortalChanged`
+        // would become the `verifyServerState` argument to `checkConnectivity`.
+        QTimer::singleShot(0, this, [this] { checkConnectivity(false); });
```

Comment:
Does it make sense to run the check if it changed to true?

Reply:
Yes, it sets the state to `Connecting` and triggers the signals so the UI shows paused and captive portal.

Reply:
Could you add that as a comment? I see the possibility that someone might "Optimize" that call.